src/idolevelmenuitem.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2023 Robert Tari
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3, as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranties of
+ * MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Robert Tari <robert@tari.in>
+ */
+
+#include "idolevelmenuitem.h"
+#include "idoactionhelper.h"
+
+enum
+{
+    PROP_0,
+    PROP_ICON,
+    PROP_TEXT,
+    PROP_LEVEL,
+    PROP_LAST
+};
+
+static GParamSpec *lProperties[PROP_LAST];
+
+typedef struct
+{
+    GIcon *pIcon;
+    gchar *sText;
+    GtkWidget *pLabel;
+    GtkWidget *pImage;
+    GtkWidget *pLevelBar;
+    IdoActionHelper *pHelper;
+
+} IdoLevelMenuItemPrivate;
```

Comment:
We need a Qt implementation for this in qmenumodel / Lomiri, right???

Reply:
No, this is for desktops. Lomiri uses the Progress Menu Item.